charts/prometheus-pushgateway/templates/service.yaml

```diff
@@ -14,7 +14,7 @@ metadata:
 spec:
   {{- if .Values.service.clusterIP }}
   clusterIP: {{ .Values.service.clusterIP }}
-  {{ else if and .Values.runAsStatefulSet (ne .Values.service.type "LoadBalancer")}}
+  {{ else if and .Values.runAsStatefulSet (and (ne .Values.service.type "LoadBalancer") (ne .Values.service.type "NodePort"))}}
```

Comment:
How about:

```suggestion
  {{ else if and .Values.runAsStatefulSet (not has .Values.service.type ["LoadBalancer", "NodePort"]) }}
```

Reply:
Thank you. The proposed change produces the _unexpected "[" in operand_ error for me. I would suggest e.g.
```
{{- $stsExcludeServiceTypes := list "LoadBalancer" "NodePort" }}

{{ else if and .Values.runAsStatefulSet (not (has .Values.service.type $stsExcludeServiceTypes)) }}
```
which behaves as expected.